### DISCUSSION TITLE
How to get updated state inside onValid callback?

### DISCUSSION THREAD

**Discussion by caio-borghi-yapi** [react-hook-form/react-hook-form#8861] [Q&A] [answered]
When the onValid callback is executed, it always use old state.

I created a wrapper to handleSubmit so I can disable a global flag before calling it (it's use to display leave without saving modal abroad the application).

But, no matter when I update the state (redux, context, or useState), when the onValid is executed, it always have the old version of the state.

Here's a sandbox where you can understand the issue.

I need to find a way to ensure handleSubmit callbacks use the latest state.

https://codesandbox.io/s/onvalid-always-old-state-4or4ie?file=/src/App.js

**Comment by bluebill1049** [comment, marked as answer]
at the moment, a state update is required to be the next render, so only `useEffect` would get the updated state.
